From d75e5a5352c27af0ec6a22b5302910166987f98c Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Sat, 25 Feb 2023 23:11:57 -0500
Subject: deps/tools: [msys2] Fix tar file creation

Add CREATE_TAR_FLAGS to avoid symbolic links under Windows build OS.
---
 deps/tools/common.mk | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/deps/tools/common.mk b/deps/tools/common.mk
index 9c204ff629..0d68d9b81e 100644
--- a/deps/tools/common.mk
+++ b/deps/tools/common.mk
@@ -163,11 +163,18 @@ reinstall-$(strip $1):
 	+$$(MAKE) stage-$(strip $1)
 	+$$(MAKE) install-$(strip $1)
 
+ifeq ($(BUILD_OS),WINNT)
+# MSys2 has issues with symbolic links
+CREATE_TAR_FLAGS := --dereference -cf
+else
+CREATE_TAR_FLAGS := -cf
+endif
+
 $$(build_staging)/$2.tar: $$(BUILDDIR)/$2/build-compiled
 	rm -rf $$(build_staging)/$2
 	mkdir -p $$(build_staging)/$2$$(build_prefix)
 	$(call $3,$$(BUILDDIR)/$2,$$(build_staging)/$2,$4)
-	cd $$(build_staging)/$2$$(build_prefix) && $$(TAR) -cf $$@.tmp .
+	cd $$(build_staging)/$2$$(build_prefix) && $$(TAR) $$(CREATE_TAR_FLAGS) $$@.tmp .
 	rm -rf $$(build_staging)/$2
 	mv $$@.tmp $$@
 
-- 
